keras/layers/embeddings.py

```diff
@@ -55,6 +55,7 @@ class Embedding(Layer):
           This argument is required if you are going to connect
           `Flatten` then `Dense` layers upstream
           (without it, the shape of the dense outputs cannot be computed).
+      output_dtype: dtype of embedding weights. Default (`None`) uses `K.floatx()`.
```

Comment:
As far as I can tell you can remove this argument and do the following:
- remove `kwargs['dtype'] = 'int32'`
- use `dtype=self.output_dtype` in `add_weight`